standard/links_stream.lua

```diff
@@ -142,11 +233,15 @@ function StreamKey:new(tbl, languageCode, index)
 		local components = mw.text.split(tbl, '_', true)
 		-- Input is in legacy format (eg. twitch2)
 		if #components == 1 then
-			platform, index = self:_fromLegacy(tbl)
+			local tempIndex
+			platform, tempIndex = self:_fromLegacy(tbl)
+			index = tempIndex or index
```

Comment:
What's the scenario this is needed? Looking at fromLegacy, either platform + index are both nil or both have values?

Reply:
letting it error instead